File: backend/app/schemas/extra_task.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ExtraTaskCreate(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str | None = None
    deadline: datetime | None = None


class ExtraTaskUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=200)
    description: str | None = None
    deadline: datetime | None = None
    position: int | None = None


class ExtraTaskRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    title: str
    description: str | None
    deadline: datetime | None
    is_completed: bool
    completed_at: datetime | None
    position: int
    is_overdue: bool

File: mobile/lib/models/extra_task.dart
class ExtraTask {
  final String id;
  final String title;
  final String? description;
  final DateTime? deadline;
  final bool isCompleted;
  final DateTime? completedAt;
  final int position;
  final bool isOverdue;

  const ExtraTask({
    required this.id,
    required this.title,
    required this.description,
    required this.deadline,
    required this.isCompleted,
    required this.completedAt,
    required this.position,
    required this.isOverdue,
  });

  factory ExtraTask.fromJson(Map<String, dynamic> json) => ExtraTask(
        id: json['id'] as String,
        title: json['title'] as String,
        description: json['description'] as String?,
        deadline: json['deadline'] == null ? null : DateTime.parse(json['deadline'] as String),
        isCompleted: json['is_completed'] as bool,
        completedAt:
            json['completed_at'] == null ? null : DateTime.parse(json['completed_at'] as String),
        position: json['position'] as int,
        isOverdue: json['is_overdue'] as bool,
      );
}
